omni-relayer/src/utils/near.rs

```diff
@@ -88,10 +88,51 @@ pub async fn handle_streamer_message(
                 )
                 .await;
             }
-            Nep141LockerEvent::FinTransferEvent { .. }
-            | Nep141LockerEvent::LogMetadataEvent { .. }
-            | Nep141LockerEvent::SignClaimNativeFeeEvent { .. }
-            | Nep141LockerEvent::ClaimFeeEvent { .. } => {}
+            Nep141LockerEvent::FinTransferEvent {
+                ref nonce,
+                ref transfer_message,
+            } => {
+                if nonce.is_none() {
+                    utils::redis::add_event(
+                        redis_connection,
+                        utils::redis::NEAR_FIN_TRANSFER_EVENTS,
+                        transfer_message.origin_nonce.0.to_string(),
+                        log,
+                    )
+                    .await;
+                }
+            }
+            Nep141LockerEvent::ClaimFeeEvent {
+                ref transfer_message,
+                ref native_fee_recipient,
+            } => {
+                if native_fee_recipient == &config.near.relayer_address_on_evm {
+                    utils::redis::add_event(
+                        redis_connection,
+                        utils::redis::NEAR_FIN_TRANSFER_EVENTS,
+                        transfer_message.origin_nonce.0.to_string(),
+                        log,
+                    )
+                    .await;
+                }
+            }
+            Nep141LockerEvent::SignClaimNativeFeeEvent {
+                ref claim_payload, ..
+            } => {
+                utils::redis::add_event(
+                    redis_connection,
+                    utils::redis::NEAR_SIGN_CLAIM_NATIVE_FEE_EVENTS,
+                    claim_payload
+                        .nonces
+                        .iter()
+                        .map(|nonce| nonce.0.to_string())
+                        .collect::<Vec<_>>()
+                        .join(","),
+                    log,
+                )
+                .await;
+            }
```

Comment:
Also, it would be useful to check the recipient to handle only our transactions.

Reply:
Fixed in fdf5afdbf258aad37a0caf64b758d9c403cf77dc